models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+call nc_get_variable(ncid, 'lat_rho', TLAT, routine)
+call nc_get_variable(ncid, 'lon_u'  , ULON, routine)
+call nc_get_variable(ncid, 'lat_u'  , ULAT, routine)
+call nc_get_variable(ncid, 'lon_v'  , VLON, routine)
+call nc_get_variable(ncid, 'lat_v'  , VLAT, routine)
+
+where (TLON < 0.0_r8) TLON = TLON + 360.0_r8
+where (ULON < 0.0_r8) ULON = ULON + 360.0_r8
+where (VLON < 0.0_r8) VLON = VLON + 360.0_r8
+
+! Be aware that all the depths are negative values.
+! The surface of the ocean is 0.0, the deepest is a big negative value.
+if (do_output() .and. debug > 0) then
+    write(string1,*) '    min/max ULON ', minval(ULON), maxval(ULON)
+    write(string2,*)     'min/max ULAT ', minval(ULAT), maxval(ULAT)
+    write(string3,*)     'min/max UDEP ', minval(UDEP, MASK=UDEP>Zm), maxval(UDEP)
+    call error_handler(E_MSG, routine, string1, text2=string2, text3=string3)
+
+    write(string1,*) '    min/max VLON ', minval(VLON), maxval(VLON)
+    write(string2,*)     'min/max VLAT ', minval(VLAT), maxval(VLAT)
+    write(string3,*)     'min/max VDEP ', minval(VDEP, MASK=VDEP>Zm), maxval(VDEP)
+    call error_handler(E_MSG, routine, string1, text2=string2, text3=string3)
+
+    write(string1,*) '    min/max TLON ', minval(TLON), maxval(TLON)
+    write(string2,*)     'min/max TLAT ', minval(TLAT), maxval(TLAT)
+    write(string3,*)     'min/max TDEP ', minval(TDEP, MASK=TDEP>Zm), maxval(TDEP)
+    call error_handler(E_MSG, routine, string1, text2=string2, text3=string3)
+endif
+
+call nc_close_file(ncid, routine)
+
+end subroutine get_grid
+
+
+!-----------------------------------------------------------------------
+! Get physical depth z_rho using ROMS formulations:
+! refer to set_depth.F
+
+subroutine compute_physical_depth(b, z, d)
+
+character(len=*), parameter :: routine = 'compute_physical_depth'
+
+real(r8), intent(in)  :: b, z   ! bathymetry and free surface
+real(r8), intent(out) :: d(Nz)  ! depth at all levels
+real(r8)              :: z0
+
+if (debug > 100) then 
+   do ik = 1, Nz
+      write(*, '(A, i3, A, f10.7)') 'Layer: ', ik, ', S_RHO: ', sr(ik) 
+   enddo
+endif
+
+! Compute z at RHO points 
+do ik = 1, Nz
+   if (Vt == 1) then ! Original transformation
+      z0    = hc*(sr(ik) - Cr(ik)) + Cr(ik)*b 
+      d(ik) = z0 + z * (1.0_r8 + z0/b)
+   elseif (Vt == 2) then ! New transformation
+      z0    = (hc*sr(ik) + Cr(ik)*b)/(hc+b) 
+      d(ik) = z + z0*(z+b) 
+   else
+      string1 = 'Unsupported Vtransform'
+      call error_handler(E_ERR, routine, string1)
+   endif 
+enddo
+
+! Reverse the sign: 
+! Below the surface is +ve depth
+! Surface is 0
+! Above the surface is -ve depth
+d = -d
+
+end subroutine compute_physical_depth
+
+
+!-----------------------------------------------------------------------
+! Using z_rho, compute z_u and z_v
+
+subroutine U_V_depths()
+
+! Compute z_u; average in xi direction 
+do ix = 1, Nx-1
+   do iy = 1, Ny
+      UDEP(ix, iy, :) = 0.5_r8 * (TDEP(ix, iy, :) + TDEP(ix+1, iy, :))
```

Comment:
if {u|v} dep is a function of only tdep then you only need to store tdep, you could calculated {u|v} depth on the fly if memory becomes and issue.